cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -373,55 +373,56 @@ void SparsePoseGraph::FreezeTrajectory(const int trajectory_id) {
   });
 }
 
-void SparsePoseGraph::AddSubmapFromProto(const int trajectory_id,
-                                         const transform::Rigid3d& initial_pose,
+void SparsePoseGraph::AddSubmapFromProto(const transform::Rigid3d& global_pose,
                                          const mapping::proto::Submap& submap) {
   if (!submap.has_submap_2d()) {
     return;
   }
 
+  const mapping::SubmapId proto_submap_id = {submap.submap_id().trajectory_id(),
```

Comment:
Conciseness: Just `submap_id`? It is not just in the proto, it is also used everywhere else.

Reply:
Sure. It's an artifact from times when we received the actual inserted id from `NestedVectorsById::Append`.